### ISSUE TITLE
Investigate: GDA should call Hyperion once per sample

### ISSUE THREAD

**Issue by DominicOram** [DiamondLightSource/mx-bluesky#202]
We believe we now have all the logic for UDC in Hyperion. This should mean the GDA just needs to call Hyperion once for a sample to do the whole collection. However, we should investigate if there is anything else GDA is doing that we have missed, particularly for cases like S+C or Chi changes.

## Acceptance Criteria
* There are detailed tickets on what more we need to add into Hyperion to get it to do the whole collection in one call

**Comment by DominicOram** [comment]
Looking in i03's HCR on master:
* Collection comes in to the `execute_request`
* On line 382 this calls hyperion's `execute_sample_loading` - we should aim to replace this with a call to hyperion that does the whole of the rest of the function
* `execute_datacollection_prepare` is then called. This firstly does centring, which we already have in hyperion.
* We then check the cryo is in - I think the robot does this anyway so it's unnecessary **confirm with scientist**
* There is then a section for plate screen mode - I think this is out of scope for hyperion **confirm with scientist**
* We then take snapshots, move the backlight out and the aperture in - hyperion does this already
* There is then some code around `spectrum_taker`, which is for metalID collections - I believe outside the scope of hyperion currently **confirm with scientist**
* Sample position is then put in the comment - hyperion already does this
* Energy/wavelength is then changed - hyperion can do this but needs stress testing
* Beam size is set - we do this already in hyperion
* Aperture added to comment - done in hyperion
* `execute_request` then checks if the current is below some minimum value and waits for it to be above that - we don't do this in hyperion, maybe we should **confirm with scientist**
* Then opens detector shutter, we do this in hyperion
* Then calls `do_sample` where we first disables ROI mode, we do this in hyperion
* Then sets transmission to 1 and locks XBPM - hyperion already does this straight after XRC
* Then some dose scaling - I don't think this is turned on for i03 but we will need it at some point **confirm with scientist**
* Then changing transmission - hyperion already does this but should do it better see https://github.com/DiamondLightSource/mx-bluesky/issues/205
* Then sets some parameters on the detector - hyperion already does this
* Then starts looping through the oscillations - I think this is always only one collection for UDC **confirm with scientist**
* Then clears odin errors, hyperion already does this
* Then triggers zocalo - hyperion already does this
* Then `restore_stub_offsets`, which we are not using for now
* Then in the finally turns xbpm feedback on again - https://github.com/DiamondLightSource/mx-bluesky/issues/205 will fix this
*

**Comment by DominicOram** [comment]
Discussed with @neil-i03 with the following conclusions:
1. Cryro in check is not required
2. Plate screening is outside the scope of hyperion but we should check that we're not accidentally doing it https://github.com/DiamondLightSource/mx-bluesky/issues/467
3. Metal ID can be ignored for now, we will come up with a proper solution to it in the future
4. The ring current check is actually only checking if the current is >0 at the moment. This should be caught by xbpm feedback anyway so no need to add this
5. There is some dose scaling on i03 currently but it's handled by Agamemnon, we do not need to worry about it.
6. Multiple oscillations are only used for stepped transmission collections, these are an edge case we can handle separately, see https://github.com/DiamondLightSource/mx-bluesky/issues/471

In conclusion to do this we will need to do, in roughly this order:
* https://github.com/DiamondLightSource/mx-bluesky/issues/205
* https://github.com/DiamondLightSource/mx-bluesky/issues/467
* https://github.com/DiamondLightSource/mx-bluesky/issues/240
* https://github.com/DiamondLightSource/mx-bluesky/issues/470
* https://github.com/DiamondLightSource/mx-bluesky/issues/471

**Comment by DominicOram** [comment]
This is now being implemented so calling the issue donw
